src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -48,6 +48,8 @@ public sealed partial class SqlConnection : DbConnection, ICloneable
         // using SqlConnection.Open() method. 
         internal bool _applyTransientFaultHandling = false;
 
+        internal SqlTransaction CurrentTransaction { get; set; }
```

Comment:
Hmm. I thought this whole isolation level "leak" thing started with system.transactions.